core-plugins/src/test/java/io/cdap/plugin/batch/joiner/JoinerTestRun.java

```diff
@@ -309,9 +310,32 @@ public void testOuterJoinWithoutRequiredInputs() throws Exception {
                                    "joinKeys", "film.film_id=filmActor.film_id=filmCategory.film_id&" +
                                      "film.film_name=filmActor.film_name=filmCategory.film_name",
                                    "selectedFields", selectedFields,
-                                   "requiredInputs", ""),
+                                   "requiredInputs", "",
+                                   "schema", outputSchema.toString()),
                                  null));
 
+    ETLStage joinSinkStage = new ETLStage(
+      "sink", new ETLPlugin("TPFSAvro", BatchSink.PLUGIN_TYPE,
+                            ImmutableMap.of(Properties.TimePartitionedFileSetDataset.SCHEMA, outputSchema.toString(),
+                                            Properties.TimePartitionedFileSetDataset.TPFS_NAME, joinedDatasetName),
+                            null));
+
+    joinHelper(filmStage, filmActorStage, filmCategoryStage, joinStage, joinSinkStage,
+               filmDatasetName, filmActorDatasetName, filmCategoryDatasetName, joinedDatasetName,
+               outputSchema, this::verifyOuterJoinWithoutRequiredInputs);
+  }
+
+  @Test
+  public void testOuterJoinWithoutRequiredInputs() throws Exception {
```

Comment:
empty required inputs means an outer join? We don't need to run a full pipeline in order to test this, it should be handled in the JoinerConfigTest with maybe some refactoring (these pipeline tests take a long time to run and are really more like integration tests).

Also, is this testing some behavior that you changed in this PR?

Reply:
this PR isn't really changing outer join behavior. I've removed `testOuterJoinWithoutRequiredInputs`.